Plutarch/Unit.hs

```diff
@@ -39,4 +38,4 @@ instance Monoid (Term s PUnit) where
   mempty = pcon PUnit
 
 instance PShow PUnit where
-  pshow' _ x = pmatch x $ \PUnit -> "()"
+  pshow' _ _ = "()"
```

Comment:
this is actually an oversight in the other instances, and I don't think this is actually a "fix", see #341